Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccadmission/storage/DccAdmissionCheckScenariosRepository.kt

```diff
@@ -0,0 +1,41 @@
+package de.rki.coronawarnapp.ccl.dccadmission.storage
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import de.rki.coronawarnapp.ccl.configuration.update.CCLSettings
+import de.rki.coronawarnapp.ccl.dccadmission.model.DccAdmissionCheckScenarios
+import de.rki.coronawarnapp.util.serialization.BaseJackson
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import timber.log.Timber
+import javax.inject.Inject
+
+@Suppress("BlockingMethodInNonBlockingContext")
+class DccAdmissionCheckScenariosRepository @Inject constructor(
+    private val cclSettings: CCLSettings,
+    @BaseJackson private val mapper: ObjectMapper
+) {
+
+    val admissionCheckScenarios: Flow<DccAdmissionCheckScenarios?> =
+        cclSettings.admissionCheckScenarios.map {
+            it?.let { json ->
+                try {
+                    mapper.readValue(json, DccAdmissionCheckScenarios::class.java)
+                } catch (e: Exception) {
+                    Timber.e(e, "Failed to parse admission check scenarios.")
+                    null
+                }
+            } ?: run {
+                Timber.v("No admission check scenarios available.")
+                null
+            }
+        }
+
+    suspend fun save(scenarios: DccAdmissionCheckScenarios) {
+        val json = mapper.writeValueAsString(scenarios)
+        cclSettings.setAdmissionCheckScenarios(json)
+    }
+
+    suspend fun clear() {
+        cclSettings.setAdmissionCheckScenarios("")
```

Comment:
Shouldn't `null` be passed when clearing them instead of `""`?

Reply:
changed it to null, but the data store seems to only accept non-null, so then there it will be an empty string